test/crowdsale/PausableCrowdsale.test.js

```diff
@@ -0,0 +1,153 @@
+        });
+
+        it('reverts', async function () {
+          await shouldFail.reverting(this.crowdsale.pause({ from }));
+        });
+      });
+    });
+
+    describe('when the sender is not the crowdsale owner', function () {
+      const from = stranger;
+
+      it('reverts', async function () {
+        await shouldFail.reverting(this.crowdsale.pause({ from }));
+      });
+    });
+  });
+
+  describe('unpause()', function () {
+    describe('when the sender is the crowdsale owner', function () {
+      const from = owner;
+
+      describe('when the crowdsale is paused', function () {
+        beforeEach(async function () {
+          await this.crowdsale.pause({ from });
+        });
+
+        it('unpauses the crowdsale', async function () {
+          await this.crowdsale.unpause({ from });
+          const paused = await this.crowdsale.paused();
+          paused.should.equal(false);
+        });
+
+        it('emits an Unpause event', async function () {
+          const { logs } = await this.crowdsale.unpause({ from });
+          expectEvent.inLogs(logs, 'Unpaused');
+        });
+      });
+
+      describe('when the crowdsale is unpaused', function () {
+        it('reverts', async function () {
+          await shouldFail.reverting(this.crowdsale.unpause({ from }));
+        });
+      });
+    });
+
+    describe('when the sender is not the crowdsale owner', function () {
+      const from = stranger;
+
+      it('reverts', async function () {
+        await shouldFail.reverting(this.crowdsale.unpause({ from }));
+      });
+    });
+  });
+
+  describe('paused', function () {
+    const from = owner;
+
+    it('is not paused by default', async function () {
+      const paused = await this.crowdsale.paused({ from });
+      paused.should.equal(false);
+    });
+
+    it('is paused after being paused', async function () {
+      await this.crowdsale.pause({ from });
+      const paused = await this.crowdsale.paused({ from });
+      paused.should.equal(true);
+    });
+
+    it('is not paused after being paused and then unpaused', async function () {
+      await this.crowdsale.pause({ from });
+      await this.crowdsale.unpause({ from });
+      const paused = await this.crowdsale.paused({ from });
+      paused.should.equal(false);
+    });
+  });
+
+  describe('when the crowdsale is paused', function () {
+    const from = owner;
+
+    beforeEach('pausing', async function () {
+      await this.crowdsale.pause({ from });
+    });
+
+    describe('high-level purchase using fallback function', function () {
+      it('reverts', async function () {
+        await shouldFail.reverting(this.crowdsale.sendTransaction({ from, value }));
+      });
+    });
+
+    describe('buyTokens()', function () {
+      it('reverts', async function () {
+        await shouldFail.reverting(this.crowdsale.buyTokens(from, { from, value }));
+      });
+    });
+  });
+
+  describe('when the crowdsale is unpaused', function () {
```

Comment:
The way I usually setup this sort of thing is by having sequential actions, e.g.:

```
context initially
  purchases work
  context when paused
    purchases stop working
    when unpaused
      purchases work again
```

I think that makes it easier to understand the intended lifecycle of the contract, and neatly exposes what the test intends to test ('this thing used to work and now it doesn't anymore').